plugins/inputs/x509_cert/x509_cert.go

```diff
@@ -57,20 +62,29 @@ func (c *X509Cert) SampleConfig() string {
 	return sampleConfig
 }
 
-func (c *X509Cert) locationToURL(location string) (*url.URL, error) {
-	if strings.HasPrefix(location, "/") {
-		location = "file://" + location
-	}
-	if strings.Index(location, ":\\") == 1 {
-		location = "file://" + filepath.ToSlash(location)
-	}
+func (c *X509Cert) sourcesToURLs() error {
+	for _, source := range c.Sources {
+		if strings.HasPrefix(source, "/") {
+			g, err := globpath.Compile(source)
+			if err != nil {
+				return fmt.Errorf("could not compile glob %v: %v", source, err)
+			}
+			c.globpaths = append(c.globpaths, g)
+		} else {
+			if strings.Index(source, ":\\") == 1 {
+				source = "file://" + filepath.ToSlash(source)
+			}
```

Comment:
I think this will prevent globs prefixed with `file://` from being compiled. How about
```suggestion
		if strings.HasPrefix(source, "file://") || strings.HasPrefix(source, "/") || strings.Index(source, ":\\") == 1 {
			// We have a file path, so glob it
			path := filepath.ToSlash(string.TrimPrefix(source, "file://"))
			g, err := globpath.Compile(path)
			if err != nil {
				return fmt.Errorf("could not compile glob %v: %v", source, err)
			}
			c.globpaths = append(c.globpaths, g)
		} else {
			// We don't have a file path, so simply store it as a URL
```
This leaves the necessity to always prefix relative paths with `file://` as otherwise we cannot distinguish them from other URLs.

Reply:
As I asked in my previous comment should we consider parsing "files://" since it's not supported by RFC 3986?
If yes, then we should also search for prefix "file://../" and "../" (that I forgot)
Also I dont have windows system on my hands, but I think there is no globbing support on windows system right?

Reply:
I guess Windows has support for globbing, but not sure... @ssoroka do you know if globbing works on Windows? Regarding the `file://` handling, I think we should support globbing there as you even have this as an example for the config option. :-)
Why do we need to check for `file://../`? It's covered already because we enter the "file" branch when prefixed with `file://`. For `../` it's a different story. As I said you _always_ need to prefix relative paths with `file://` otherwise you run into the trouble of checking `./something`, `somepath/` etc.